src/common/IColor.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2024, Martin Blicha <martin.blicha@gmail.com>
+ *
+ *  SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef ICOLOR_H
+#define ICOLOR_H
+
+#include <cassert>
+#include <stdexcept>
+#include <string>
+
+namespace opensmt {
+
+enum class icolor_t : char { I_UNDEF = 0, I_A = 1, I_B = 2, I_AB = I_A | I_B, I_S = 4 };
```

Comment:
Just partitioning of a formula into two parts.
I mean, this is a helper that is used when dealing with `ipartitions_t` in some context. But they are independent.

Reply:
What about adding a short comment for documentation? Since the file is placed at `common`, the reader has probably no idea what A, B, S means.

Reply:
Added a short comment.
`S` should generally be ignored. It is there because of hack that ideally should not be needed, but I don't know how to solve the issue that arises otherwise.